src/components/ChooseLanguage.tsx

```diff
@@ -0,0 +1,103 @@
+import { createForm } from "@modular-forms/solid";
+import { useNavigate } from "@solidjs/router";
+import { createSignal, For, Show } from "solid-js";
+
+import { Button, ExternalLink, InfoBox, NiceP, VStack } from "~/components";
+import { useI18n } from "~/i18n/context";
+import { useMegaStore } from "~/state/megaStore";
+import { eify, EN_OPTION, Language, LANGUAGE_OPTIONS, timeout } from "~/utils";
+
+type ChooseLanguageForm = {
+    selectedLanguage: string;
+};
+
+const COMBINED_OPTIONS: Language[] = [EN_OPTION, ...LANGUAGE_OPTIONS];
+
+export function ChooseLanguage() {
+    const i18n = useI18n();
+    const [error, setError] = createSignal<Error>();
+    const [state, actions] = useMegaStore();
+    const [loading, setLoading] = createSignal(false);
+    const navigate = useNavigate();
+
+    function findLanguageByValue(value: string) {
+        return (
+            COMBINED_OPTIONS.find((language) => language.value === value) ??
+            EN_OPTION
+        );
+    }
+
+    const [_chooseLanguageForm, { Form, Field }] =
+        createForm<ChooseLanguageForm>({
+            initialValues: {
+                selectedLanguage: findLanguageByValue(state.lang ?? "").value
+            },
+            validate: (values) => {
+                const errors: Record<string, string> = {};
+                if (values.selectedLanguage === undefined) {
+                    errors.selectedLanguage = i18n.t(
+                        "settings.language.error_unsupported_language"
+                    );
+                }
+                return errors;
+            }
+        });
+
+    const handleFormSubmit = async (f: ChooseLanguageForm) => {
+        setLoading(true);
+        const lang = findLanguageByValue(f.selectedLanguage);
+        try {
+            actions.saveLanguage(lang.value);
+
+            await i18n.changeLanguage(lang.shortName);
```

Comment:
the lang state and i18n api call should be the same key, that is why in the megaStore we are initializing with `localStorage.getItem('i18nextLng')`. since we are using the `shortName as the value we can just point to the selectedLanguage
```
actions.saveLanguage(f.selectedLanguage);

await i18n.changeLanguage(f.selectedLanguage);
```